Public/Src/Engine/Dll/Distribution/Grpc/ClientConnectionManager.cs

```diff
@@ -43,11 +45,34 @@ public ClientConnectionManager(LoggingContext loggingContext, string ipAddress,
                     port,
                     ChannelCredentials.Insecure,
                     DefaultChannelOptions);
+            m_monitorConnectionTask = MonitorConnectionAsync();
+        }
+
+        public async Task MonitorConnectionAsync()
+        {
+            await Task.Yield();
+
+            ChannelState state = ChannelState.Idle;
+            while (state != ChannelState.Shutdown)
+            {
+                await Channel.WaitForStateChangedAsync(state);

```

Comment:
>WaitForStateChangedAsync [](start = 30, length = 24)

Do we trust this to not get into a hot loop?

Reply:
yes. tested in many builds

---
In reply to: [285705092](https://github.com/microsoft/BuildXL/pull/333#discussion_r285705092) [](ancestors = 285705092)